test/base/MockAuthAction.scala

```diff
@@ -96,67 +109,13 @@ trait MockAuthAction extends MockitoSugar with MetricsMocks {
-                      new ~(
-                        new ~(
-                          new ~(
-                            new ~(
-                              new ~(
-                                new ~(
-                                  new ~(
-                                    new ~(
-                                      new ~(
-                                        new ~(
-                                          new ~(
-                                            new ~(
-                                              new ~(
-                                                user.identityData.credentials,
-                                                user.identityData.name
-                                              ),
-                                              user.identityData.email
-                                            ),
-                                            user.identityData.externalId
-                                          ),
-                                          user.identityData.internalId
-                                        ),
-                                        user.identityData.affinityGroup
-                                      ),
-                                      user.enrolments
-                                    ),
-                                    user.identityData.agentCode
-                                  ),
-                                  user.identityData.confidenceLevel.get
-                                ),
-                                user.identityData.nino
-                              ),
-                              user.identityData.saUtr
-                            ),
-                            user.identityData.dateOfBirth
-                          ),
-                          user.identityData.agentInformation.get
-                        ),
-                        nrsGroupIdentifierValue
-                      ),
-                      nrsCredentialRole
-                    ),
-                    Some(nrsMdtpInformation)
-                  ),
-                  Some(nrsItmpName)
-                ),
-                nrsDateOfBirth
-              ),
-              Some(nrsItmpAddress)
-            ),
-            nrsCredentialStrength
-          ),
-          nrsLoginTimes
-        )
+        user.identityData.credentials ~~ user.identityData.name ~~ user.identityData.email ~~
+          user.identityData.externalId ~~ user.identityData.internalId ~~ user.identityData.affinityGroup ~~
+          user.enrolments ~~ user.identityData.agentCode ~~ user.identityData.confidenceLevel.get ~~
+          user.identityData.nino ~~ user.identityData.saUtr ~~ user.identityData.dateOfBirth ~~
+          user.identityData.agentInformation.get ~~ nrsGroupIdentifierValue ~~ nrsCredentialRole ~~
+          Some(nrsMdtpInformation) ~~ Some(nrsItmpName) ~~ nrsDateOfBirth ~~ Some(nrsItmpAddress) ~~
+          nrsCredentialStrength ~~ nrsLoginTimes
```

Comment:
I have no idea what these tilda operators actually do, never mind the double tilda 😁 . I presume what you've done works and hasn't broken any tests in which case 👍 .

Reply:
the tilda class is just the auth libaries way of constructing a tuple like object, double tilda is just a method i made to tidy up the christmas tree of doom that was there, happy for suggestion of a new name if it helps clear it up. But yeah its just syntactic sugar